### ISSUE TITLE
DICUI doesn't work with mono or wine on Linux

### ISSUE THREAD

**Issue by ajshell1** [SabreTools/DICUI#135]
TL;DR
DICUI doesn't work on Linux. The latest test versions of DIC actually do work rather well, so I'd like to see DICUI ported as well.

Longer version:

I'm using the latest version of Antergos with mono 5.16.0, Wine-Staging 4.0-RC4. I also have wine-mono or whatever the package is called installed.

Results of attemping to run DICUI (using most recent appveyor build)

"mono DICUI.exe":

The entry point method could not be loaded due to Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.

"wine DICUI.exe":

The entry point method could not be loaded due to Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. assembly:PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35 type:<unknown type> member:<none>

sudo does nothing

Attempting to run the WinForms version results in an error message that is significantly larger.

https://pastebin.com/YRgXhGWw

If you won't fix this, I'd like a command line version of DICUI that just produces a !submissioninfo.txt file

**Comment by ajshell1** [comment]
Update: Installing the latest version of .NET offered by WineTricks actually makes the window show up in the WinForms build, but it gives an unhandled exception: 

See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NotImplementedException: The method or operation is not implemented.
   at System.Windows.Forms.UnsafeNativeMethods.ITextRange.ScrollIntoView(Int32 value)
   at System.Windows.Forms.TextBoxBase.ScrollToCaret()
   at DICUI.Forms.Windows.LogWindow.OnTextChanged(Object sender, EventArgs e) in C:\projects\dicui\DICUI.Forms\Windows\LogWindow.cs:line 375
   at System.Windows.Forms.Control.OnTextChanged(EventArgs e)
   at System.Windows.Forms.TextBoxBase.OnTextChanged(EventArgs e)
   at System.Windows.Forms.TextBoxBase.WmReflectCommand(Message& m)
   at System.Windows.Forms.TextBoxBase.WndProc(Message& m)
   at System.Windows.Forms.RichTextBox.WmReflectCommand(Message& m)
   at System.Windows.Forms.RichTextBox.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.NET/Framework/v4.0.30319/mscorlib.dll
----------------------------------------
DICUI.Forms
    Assembly Version: 1.11.0.0
    Win32 Version: 1.11.0.0
    CodeBase: file:///Z:/home/aj/software/DICUI.Forms/bin/Debug/DICUI.Forms.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
DICUI.Library
    Assembly Version: 1.11.0.0
    Win32 Version: 1.11.0.0
    CodeBase: file:///Z:/home/aj/software/DICUI.Forms/bin/Debug/DICUI.Library.DLL
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
WindowsBase
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3062.0 built by: NET472REL1
    CodeBase: file:///C:/windows/Microsoft.Net/assembly/GAC_MSIL/WindowsBase/v4.0_4.0.0.0__31bf3856ad364e35/WindowsBase.dll
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.

**Comment by mnadareski** [comment]
Leaving this as a "Won't Fix" for now since mono is stupid sometimes, and neither UI framework we use for DICUI is fully supported.

**Comment by mnadareski** [comment]
This has somewhat been alleviated by @ajshell1 's own script which wraps DICUI-Check. Not entirely working the same, but close.

**Comment by mnadareski** [comment]
I'm closing this in lieu of the other open issue for porting to .NET Core whenver WPF is supported (if that ever happens). I doubt this would ever get properly working on Mono or Wine.
